bundles/org.palladiosimulator.simexp.core/src/org/palladiosimulator/simexp/core/valuefunction/ValueFunctionEstimator.java

```diff
@@ -0,0 +1,8 @@
+package org.palladiosimulator.simexp.core.valuefunction;
+
+import org.palladiosimulator.simexp.core.evaluation.SampleModelIterator;
+
+public interface ValueFunctionEstimator {
+	
+	public ValueFunction estimate(SampleModelIterator iterator);

```

Comment:
Ich wäre dafür, dass als Parameter nicht der Iterator sondern direkt List<SimulatedExperience> traj übergeben wird. Schärft das Verständnis über welche Entität geschätzt werden soll. SampleModelIterator drückt dass nicht offensichtlich aus und macht somit das Verständnis des Interfaces schwierig. Habe es erst nach dem Blick auf dei Implementierung verstandn

Reply:
Der SampleModelIterator entspricht dem Implementierngskonzept welches über das sample model iteriert bzw. die einzelnen Trajektorien zurückgibt; deshalb, finde ich, ist die Semantik hier eindeutig, oder? Sprich: Die estimate-Methode bekommt das sample model (in Form eines Iterators) rein und liefer die geschätzte Value-Function zurück. Darüber hinaus: Wenn ich nur die Liste übergebe, dann muss ich die mir erst mal aus dem SampleModelIterator konstruieren, sprich: ich muss einmal komplett durch iterieren nur um die Liste zu erhalten; das ist ineffizient.

Reply:
Sie ist nicht wirklich eindeutig; Das Iterieren über das Modell und somit die Bereitstellung von Modellelementen ist eine andere Aufgabe wie das Interpretieren/Verarbeiten von einzelnen Elementen des Modells.
Somit gehören diese 2 Tätigkeiten eigentlich getrennt. 1 Klasse die Modellelemente bereitstellt und 1 Klasse die diese bereitgestellten Modellelemente dann verarbeitet. Wie effizient die Bereitstellung der Modelle ausimplementiert ist, sollte dann die ProviderKlasse selber entscheiden. Aber für die Verarbeitung ist dieser Punkt irrelevant.